File: demo/ChengFenStore/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using ChengFenStore.Services;
using ChengFenStore.Models;
using Microsoft.AspNetCore.Authorization;

namespace ChengFenStore.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class UserController : ControllerBase
    {
        private readonly IUserService _userService;

        public UserController(IUserService userService)
        {
            _userService = userService;
        }

        [AllowAnonymous]
        [HttpPost("register")]
        public IActionResult Register(UserRegistrationRequest request)
        {
            var result = _userService.Register(request);
            if (result.Success)
            {
                return Ok(result);
            }
            return BadRequest(result);
        }

        [AllowAnonymous]
        [HttpPost("login")]
        public IActionResult Login(UserLoginRequest request)
        {
            var result = _userService.Login(request);
            if (result.Success)
            {
                return Ok(result);
            }
            return Unauthorized(result);
        }

        [Authorize]
        [HttpGet("profile")]
        public IActionResult GetUserProfile()
        {
            var userId = User.Identity.Name;
            var userProfile = _userService.GetUserProfile(userId);
            if (userProfile == null)
            {
                return NotFound();
            }
            return Ok(userProfile);
        }
    }
}

File: demo/ChengFenStore/controllers/user_controller.py
from flask import Blueprint, request, jsonify
from services.user_service import UserService

user_bp = Blueprint('user', __name__)

@user_bp.route('/register', methods=['POST'])
def register_user():
    data = request.get_json()
    phone_number = data.get('phone_number')
    name = data.get('name')
    password = data.get('password')
    verification_code = data.get('verification_code')
    result = UserService.register(phone_number, name, password, verification_code)
    return jsonify(result)

@user_bp.route('/login', methods=['POST'])
def login_user():
    data = request.get_json()
    phone_number = data.get('phone_number')
    password = data.get('password')
    verification_code = data.get('verification_code')
    result = UserService.login(phone_number, password, verification_code)
    return jsonify(result)

@user_bp.route('/profile', methods=['GET'])
def get_user_profile():
    phone_number = request.args.get('phone_number')
    result = UserService.get_user_profile(phone_number)
    return jsonify(result)
